packages/fuel-indexer-schema/src/db/tables.rs

```diff
@@ -428,8 +421,89 @@ impl SchemaBuilder {
                     virtual_columns: Vec::new(),
                 });
             }
+            TypeKind::Union(u) => {
+                // We process this type effectively the same as we process `TypeKind::Object`.
+                //
+                // Except instead of using the `FieldDefinition` provided by the parser, we manually
+                // construct the `FieldDefinition` based on the fields of the union members.
+                self.parsed_schema.union_names.insert(typ.name.to_string());
+
+                self.types.insert(typ.name.to_string());
+
+                self.parsed_schema
+                    .parsed_type_names
+                    .insert(typ.name.to_string());
+
+                let field_defs = u
+                    .members
+                    .iter()
+                    .flat_map(|m| {
+                        let name = m.node.to_string();
+                        self.parsed_schema
+                            .object_field_mappings
+                            .get(&name)
+                            .unwrap_or_else(|| {
+                                panic!(
+                                    "Could not find union member '{name}' in the schema."
+                                );
+                            })
+                            .iter()
+                            .map(|(k, v)| (k.to_owned(), v.to_owned()))
+                    })
+                    .collect::<LinkedHashSet<(String, String)>>()
```

Comment:
@deekerno Explanation:
- We need a set here because we only need the unique fields of each union member
- We need to maintain order here because if we don't, then each time you hit this codepath, your fields might be in a different order
  - So you might've compiled your indexer with the fields in a certain order, then when you load the columns from the DB, they will be put in a different order (which will throw errors when you try to insert data)
- I saw that `BtreeMap` was good for key->value order, but I just needed a flat set here and [this SO answer](https://stackoverflow.com/a/46934869) seemed OK
  - The crate seems pretty stable so I chose to go with it

Reply:
Sounds good. 👍🏽